graph/db/graph.go

```diff
@@ -159,3 +161,71 @@ func (c *ChannelGraph) DisconnectBlockAtHeight(height uint32) (
 
 	return edges, nil
 }
+
+// PruneGraph prunes newly closed channels from the channel graph in response
+// to a new block being solved on the network. Any transactions which spend the
+// funding output of any known channels within he graph will be deleted.
+// Additionally, the "prune tip", or the last block which has been used to
+// prune the graph is stored so callers can ensure the graph is fully in sync
+// with the current UTXO state. A slice of channels that have been closed by
+// the target block are returned if the function succeeds without error.
+func (c *ChannelGraph) PruneGraph(spentOutputs []*wire.OutPoint,
+	blockHash *chainhash.Hash, blockHeight uint32) (
+	[]*models.ChannelEdgeInfo, error) {
+
+	c.cacheMu.Lock()
+	defer c.cacheMu.Unlock()
+
+	edges, err := c.KVStore.PruneGraph(spentOutputs, blockHash, blockHeight)
+	if err != nil {
+		return nil, err
+	}
+
+	// Now that the graph has been pruned, we'll also attempt to
+	// prune any nodes that have had a channel closed within the
+	// latest block.
+	nodes, err := c.KVStore.PruneGraphNodes()
```

Comment:
i think `KVStore.PruneGraph` should also just call `KVStore.PruneGraphNodes()` in the same tx both so that pruning remains atomic on the store level